ddtrace/debugging/_uploader.py

```diff
@@ -84,9 +84,9 @@ def upload(self) -> None:
 
     def periodic(self) -> None:
         """Upload the buffer content to the logs intake."""
-        count = self._encoder.count
+        count = self._queue.count
         if count:
-            payload = self._encoder.encode()
+            payload = self._queue.flush()
             if payload is not None:
                 try:
                     self._write_with_backoff(payload)
```

Comment:
I find that the implementation of _uploader and _probeStatusLogger while using similar terms and design have different details that make it hard to see. for example _probeStatusLogger use _write_payload_with_backoff that is called inside the flush while this class use flush to only generate payload and then call _write_with_backoff (different name).

I don't think this need to be on this PR but maybe we can look into simplify those two classes?